src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs

```diff
@@ -293,8 +294,39 @@ public Collection<AttachmentSet> TestCaseEnded(TestCaseEndEventArgs testCaseEndE
 
             this.SendEvent(testCaseEndEventArgs);
 
-            // todo : get actual test casel leve attachments here.
-            return new Collection<AttachmentSet>();
+            List<AttachmentSet> result = null;
+            try
+            {
+                result = this.attachmentManager.GetAttachments(testCaseEndEventArgs.Context);
+            }
+            catch (Exception ex)
+            {
+                if (EqtTrace.IsErrorEnabled)
+                {
+                    EqtTrace.Error("DataCollectionManager.TestCaseEnded: Failed to get attachments : {0}", ex.Message);
+                }
+
+                return new Collection<AttachmentSet>(result);
```

Comment:
`this.attachmentManager.GetAttachments` will always return an empty string. Therefore, haven't created another empty list here.